packages/std/src/math/int256.rs

```diff
@@ -0,0 +1,1202 @@
+        }
+        self.0
+            .checked_div_euclid(other.0)
+            .map(Self)
+            .ok_or(DivisionError::Overflow)
+    }
+
+    pub fn checked_rem(self, other: Self) -> Result<Self, DivideByZeroError> {
+        self.0
+            .checked_rem(other.0)
+            .map(Self)
+            .ok_or_else(|| DivideByZeroError::new(self))
+    }
+
+    pub fn checked_shr(self, other: u32) -> Result<Self, OverflowError> {
+        if other >= 256 {
+            return Err(OverflowError::new(OverflowOperation::Shr, self, other));
+        }
+
+        Ok(Self(self.0.shr(other)))
+    }
+
+    pub fn checked_shl(self, other: u32) -> Result<Self, OverflowError> {
+        if other >= 256 {
+            return Err(OverflowError::new(OverflowOperation::Shl, self, other));
+        }
+
+        Ok(Self(self.0.shl(other)))
+    }
+
+    pub fn checked_neg(self) -> Result<Self, OverflowError> {
+        self.0
+            .checked_neg()
+            .map(Self)
+            .ok_or_else(|| OverflowError::new(OverflowOperation::Neg, self, self))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    #[inline]
+    pub fn wrapping_add(self, other: Self) -> Self {
+        Self(self.0.wrapping_add(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    #[inline]
+    pub fn wrapping_sub(self, other: Self) -> Self {
+        Self(self.0.wrapping_sub(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    #[inline]
+    pub fn wrapping_mul(self, other: Self) -> Self {
+        Self(self.0.wrapping_mul(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    #[inline]
+    pub fn wrapping_pow(self, other: u32) -> Self {
+        Self(self.0.wrapping_pow(other))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_add(self, other: Self) -> Self {
+        Self(self.0.saturating_add(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_sub(self, other: Self) -> Self {
+        Self(self.0.saturating_sub(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_mul(self, other: Self) -> Self {
+        Self(self.0.saturating_mul(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_pow(self, exp: u32) -> Self {
+        Self(self.0.saturating_pow(exp))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn abs_diff(self, other: Self) -> Uint256 {
+        Uint256(self.0.abs_diff(other.0))
+    }
+}
+
+impl From<Uint256> for Int256 {
```

Comment:
Should we really implement unsingned -> signed of the same length?

I think the standard library only implements the safe ones and then has e.g. `impl TryFrom<u128> for i128`

Reply:
No we should not. I removed it.